ComSemApp/student/views.py

```diff
@@ -696,7 +696,13 @@ def get_context_data(self, **kwargs) -> dict[str, Any]:
         context['problems'] : list[dict[str, str | float]] = {}
         context['student'] : int = self.student.id
 
-        session : SpeakingPracticeSession = SpeakingPracticeSession.objects.filter(student=self.student).latest('date')
+        session : SpeakingPracticeSession = None
+        # This try-except is necessary because latest() will throw an exception if the queryset is empty
+        try:
+            session = SpeakingPracticeSession.objects.filter(student=self.student).latest()
+        except ObjectDoesNotExist:
```

Comment:
Doesn't every speaking practice need to create a new session?

Reply:
No. It's possible that a student creates a session, but closes the page before any attempts are created. We don't want empty sessions sitting around. However, we need to change the time on old sessions so they reflect the current time.